sysvabi64/sysvabi64.rst

```diff
@@ -1592,6 +1592,140 @@ IFUNC requirements for dynamic linkers
 To resolve an ``R_AARCH64_IRELATIVE`` relocation the dynamic linker
 performs the calculation described in AAELF64_ Dynamic Relocations.
 
+Function Multi-Versioning
+-------------------------
+
+Function Multi-Versioning (FMV) is an Arm C Language Extension that
+lets the compiler generate multiple function versions and auto-dispatch
+between them. Each of the function versions is specialized for a set
+of architecture extensions. The most suitable version is selected
+dynamically. This requires runtime information about the CPU features
+available on the host. FMV is supported on GNU/Linux, Android, Windows
+and many of the BSD operating systems, including Darwin.
+
+The initialization of the runtime is platform dependent. It may be
+performed at startup by calling a constructor (Windows), or it may be
+performed by calling ``__init_cpu_features_resolver`` (otherwise).
+On GNU/Linux this function is called at load time and has the following
+prototype:
+
+.. code-block:: c
+
+   void __init_cpu_features_resolver (uint64_t, const uint64_t *);
+
+The above interface expects the same parameters as a GNU Indirect
+Function resolver. Other platforms may use a different interface
+with the runtime library. However, all implementations must provide
+a DSO local definition of the function by setting the symbol
+visibility to ``STV_HIDDEN``.
+
+The runtime initialization must set a global variable which contains
+the bits corresponding to the CPU features that have been detected:
+
+.. code-block:: c
+
+   uint64_t __aarch64_cpu_features
+
+The variable may contain the following fields:
+
+.. table:: CPU features detected
+
```

Comment:
Just an observation on the implentation. I see that ``__init_cpu_features_resolver`` assumes that `__aarch64_cpu_features` has been calculated if it is non-zero.

Is there a CPU that would have none of these features set after running ``__init_cpu_features_resolver`` i.e. that function would return 0? It may be worth reserving a value that is always set so you can tell the difference between an explicit 0 and the variable is unitialized.